Examples/test-suite/ruby/preproc_runme.rb

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env ruby
+#
+# This test implementation is directly derived from its Python counterpart.
+#
+
+require 'swig_assert'
+
+# helper class for comparing version strings
+class Version
+  attr_reader :array
+  def initialize(s)
+    @array = s.split('.').map { |e| e.to_i }
+  end
+  def <(rhs)
+    a = @array.clone
+    b = rhs.array.clone
+    a << 0 while a.size < b.size
+    b << 0 while b.size < a.size
+    (a <=> b) < 0
+  end
+end
+
+if Object.const_defined?(:Warning) && Warning.respond_to?(:warn)
+  # suppressing warnings like this only works for Ruby 2.4 and later
+  module CustomWarningFilter
+    def warn(*args)
+      msg = args[0]
+      if msg =~ /[Aa]lready initialized constant Preproc::A[56]/ ||
+         msg =~ /invalid name .?__GMP_HAVE_/
+        # ignore
+      else
+        super
+      end
+    end
+  end
+  Warning.extend CustomWarningFilter
+end
```

Comment:
What is going on in the lines above? I'm just not sure what this is about!

Reply:
The extension to the `Warning` class is intended for suppressing expected Ruby warning messages about invalid or redefined Ruby constants - basically the equivalent of `%warnfilter(SWIGWARN_RUBY_WRONG_NAME)` but for the moment the wrapper library is loaded by the Ruby interpreter. I will add this very description to this section in the test case.

However now that I take a second look at this file I realize that I forgot to remove the `Version` helper class originally used for checking the Ruby version, before replacing it with the much better check for the `Warning` module.

I will commit these two small changes (comment + remove `Version`) in a moment.